addons/test_mail/tests/test_mail_activity.py

```diff
@@ -29,6 +29,79 @@ def setUpClass(cls):
         ])
 
 
+@tests.tagged('mail_activity')
+class TestActivitySearch(TestActivityCommon):
+
+    def test_search_on_date_done_includes_archived(self):
+        """
+        Test that searching on 'date_done' on mail.activity includes archived (done)
+        activities, as implemented in the custom _search method.
+        """
+        active_activity = self.env['mail.activity'].create({
+            'res_model_id': self.env['ir.model']._get_id('mail.test.activity'),
+            'res_id': self.test_record.id,
+            'user_id': self.user_employee.id,
+            'activity_type_id': self.env.ref('mail.mail_activity_data_todo').id,
+            'summary': 'Active To-Do',
+            'date_deadline': '2025-09-01',
+        })
+
+        done_activity = self.env['mail.activity'].create({
+            'res_model_id': self.env['ir.model']._get_id('mail.test.activity'),
+            'res_id': self.test_record.id,
+            'user_id': self.user_employee.id,
+            'activity_type_id': self.env.ref('mail.mail_activity_data_todo').id,
+            'summary': 'Done To-Do',
+            'date_deadline': '2025-09-02',
+        })
+        done_activity.action_done()
+
+        archived_record = self.env['mail.test.activity'].create({'name': 'Archived Test Record', 'active': False})
+        done_activity_on_archived_record = self.env['mail.activity'].create({
+            'res_model_id': self.env['ir.model']._get_id('mail.test.activity'),
+            'res_id': archived_record.id,
+            'user_id': self.user_employee.id,
+            'activity_type_id': self.env.ref('mail.mail_activity_data_todo').id,
+            'summary': 'Done To-Do on Archived Record',
+            'date_deadline': '2025-09-03',
+        })
+        done_activity_on_archived_record.action_done()
+
+        self.assertEqual(self.env['mail.activity'].search([]), active_activity)
+
+        found_activities = self.env['mail.activity'].search([('date_done', '=', fields.Date.today())])
+        self.assertIn(done_activity, found_activities)
+        self.assertIn(done_activity_on_archived_record, found_activities)
```

Comment:
search result should be consistent across runs so you can just `assertEqual(found_activities, done_activity + done_activity_on_archived_record)`
(and skip the length check)

Would say globally it's best to check exact equality in tests when not too expensive.